packages/vite/src/modes/global/build.ts

```diff
@@ -226,13 +227,31 @@ export function GlobalModeBuildPlugin(ctx: UnocssPluginContext<VitePluginConfig>
         const result = await generateAll()
         const mappedVfsLayer = Array.from(vfsLayers).map(layer => layer === LAYER_MARK_ALL ? layer : layer.replace(/^_/, ''))
         const importStatements = result.getLayer(LAYER_IMPORTS)
-        const cssWithLayers = Array.from(vfsLayers).map(layer => `${importStatements ?? ''}#--unocss-layer-start--${layer}--{start:${layer}} ${layer === LAYER_MARK_ALL
-          ? result.getLayers(undefined, [...mappedVfsLayer, LAYER_IMPORTS])
-          : (result.getLayer(layer.replace(/^_/, '')) || '')
-        } #--unocss-layer-end--${layer}--{end:${layer}}`).join('')
-
         const fakeCssId = `${viteConfig.root}/${chunk.fileName}-unocss-hash.css`
-        const css = await applyCssTransform(cssWithLayers, fakeCssId, options.dir, this)
+
+        const cssWithLayers = await Promise.all(Array.from(vfsLayers).map(async (layer) => {
+          const layerStart = `#--unocss-layer-start--${layer}--{start:${layer}}`
+          const layerEnd = `#--unocss-layer-end--${layer}--{end:${layer}}`
+
+          let layerContent
+          if (layer === LAYER_MARK_ALL) {
+            layerContent = result.getLayers(undefined, [...mappedVfsLayer, LAYER_IMPORTS])
+          }
+          else {
+            layerContent = result.getLayer(layer.replace(/^_/, '')) || ''
+          }
+
+          if (layer === LAYER_PREFLIGHTS || layer === LAYER_MARK_ALL) {
```

Comment:
While I understand that the ALL layer would contain preflights, but I am not sure if passing the whole generated CSS back to transformers is a good idea (for both performance and correctness concern).

I am not even sure if https://github.com/unocss/unocss/issues/4167 is a good idea to transform the preflights. But if we absolutely need it, maybe we should apply that in the preflight generation (before the layers been generated). But again I am not super sure about using directives in the preflights yet.